loader/config.c

```diff
@@ -5,878 +5,876 @@
-                s->expecting_end_of_value = true;
-                continue;
-            }
+    if (indent != lvl) {
+        if (indent > lvl)
+            return cfg_raise(parser, &tok.pos, "Invalid indentation") - 1;
 
-            s->open_quote_character = c;
-            continue;
+        cfg_unfetch_token(parser, &tok);
+        parser->preserve_line = false;
+        return -2 - indent;
+    }
 
-        case '[':
-            if (s->current_whitespace_depth)
-                PARSE_ERROR("loadable entry title must start on a new line");
+    if (tok.type != TK_INDENT)
+        cfg_unfetch_token(parser, &tok);
 
-            if (IS(STATE_NORMAL)) {
-                if (s->expecting_depth_plus_one)
-                    PARSE_ERROR("empty objects are not allowed");
+    parser->preserve_line = true;
 
-                if (s->within_loadable_entry && !s->consumed_at_least_one_kv)
-                    PARSE_ERROR("empty loadable entries are not allowed");
+    if (!cfg_fetch_token(parser, &tok))
+        return -1;
 
-                set_state(s, STATE_LOADABLE_ENTRY_TITLE);
-                continue;
-            }
+    if (tok.type != TK_IDENT)
+        return cfg_raise(parser, &tok.pos, "Expected identifier") - 1;
 
-            if (IS(STATE_VALUE) && s->open_quote_character) {
-                sv_extend_by(&s->current_value_view, 1);
-                continue;
-            }
+    ent = cfg_fetch_entry(parser);
 
-            PARSE_ERROR("invalid character");
-        case ']':
-            if (IS(STATE_LOADABLE_ENTRY_TITLE)) {
-                size_t prev_offset, offset;
-                s->current.t = CONFIG_ENTRY_LOADABLE_ENTRY;
-                s->current.key = s->current_value_view;
-                s->current.as_offset_to_next_loadable_entry = 0;
-
-                if (!config_emplace_entry(cfg, &s->current, &offset))
-                    PARSE_ERROR("out of memory");
-
-                if (!cfg->first_loadable_entry_offset)
-                    cfg->first_loadable_entry_offset = offset + 1;
-
-                if (cfg->last_loadable_entry_offset) {
-                    size_t prev = cfg->last_loadable_entry_offset - 1;
-                    struct config_entry *entry = &cfg->buffer[prev];
-                    entry->as_offset_to_next_loadable_entry = offset - prev;
-                }
-
-                prev_offset = s->depth_to_offset[0];
-                if (prev_offset) {
-                    struct config_entry *entry = &cfg->buffer[prev_offset - 1];
-                    entry->offset_to_next_within_same_scope = offset - (prev_offset - 1);
-                }
-                s->depth_to_offset[0] = offset;
-
-                cfg->last_loadable_entry_offset = offset + 1;
-                set_state(s, STATE_NORMAL);
-                s->expecting_end_of_value = true;
-                continue;
-            }
+    if (!ent)
+        return cfg_raise(parser, &tok.pos, "Out of config entries") - 1;
```

Comment:
Maybe a better message would be simply `Out of memory` or something like that? `Out of config entries` sounds like the user did something wrong.